### DISCUSSION TITLE
OLLAMA support

### DISCUSSION THREAD

**Discussion by fxrobin** [langchain4j/langchain4j#239] [Ideas]
LangChain supports Ollama but langchain4j does not up to now.
I feel like ollama is a great IA engine, so please when its support will be available?

Furthermore, I can contribute if there is an "how to contribute" document and a technical overview of langchain4java.

Best regards.

**Comment by cverdoes** [comment]
I just noticed that initial support for OLLAMA was merged. The OLLAMA API has a concept referred to as "context", a part of the response which can be fed back to subsequent requests to maintain a chat memory of sorts. Does this conflict with the langchain4j "ChatMemory" concept? The javadoc states that "Since language models do not keep the state of the conversation, it is necessary to provide all previous messages on every interaction with the language model". Does this mean that we should avoid adding state (such as the OLLAMA context) to (Chat)LanguageModels?

**Comment by aubeda** [comment]
First of all, fantastic work, congratulations :-) secondly I have a doubt , recently, ollama also supports the tools, this implementation is already available in LangChain, it would be interesting to include this functionality in LangChain4j, is there any work on it yet?

https://python.langchain.com/v0.2/docs/integrations/chat/ollama_functions/

**Reply by aubeda** [reply]
We are currently testing with LangChain in Python, and it works well. Additionally, the Mistral v3 model already supports functions, and even though it’s not explicitly mentioned in Llama3's documentation, we have tested it, and it works too, 

We are also trying to reproduce the same use case that we have in LangChain4j with OpenAI but using LangChain with Ollama. According to the specifications, the interface is exactly the same as OpenAI's.

https://ollama.com/library/mistral

**Reply by ch8n** [reply]
Tools are now supported in ollama will it work out of box now or require implementation?

https://ollama.com/blog/tool-support

**Reply by jaldaadithya** [reply]
Excellent! As the #1558 is merged. when can we expect it to be on the maven repo? so that end users can use this awesome feature.
